airflow/operators/docker_operator.py

```diff
@@ -169,10 +202,78 @@ def __init__(
         self.shm_size = shm_size
         if kwargs.get('xcom_push') is not None:
             raise AirflowException("'xcom_push' was deprecated, use 'BaseOperator.do_xcom_push' instead")
+        if not container_config:
+            self.container_config = {"user": self.user, "working_dir": self.working_dir,
+                                     "image": self.image}
```

Comment:
Doesn't the fact that you add `"image"` in your dict shortcuts the behavior in `__sanitize_container_config()`

```
if 'image' not in self.container_config and self.image is None:
            raise AirflowException('No docker image provided')
```

Because you set it earlier, you will never fall into the `AirflowException`

Reply:
Actually nevermind ! I see one case where you would fall in the Exception is if you have defined your `container_config` without the `"image"` and that you have not defined the image as a kwarg ! my bad !